### DISCUSSION TITLE
Download/copy logs

### DISCUSSION THREAD

**Discussion by tvd0x2a** [craftcms/cloud#24] [Feature Requests & Enhancements]
The logs UX for live environments and builds is pretty challenging to use. The window is really small—it doesn't fill the entire available space—and it's impossible to search the build logs.

Today, I very meticulously copied 2,203 lines of logs from the deployment window. This was hard because the scrolling works against the highlight all operation. Could there be a download or copy all logs option? Or something similarly simple?

I'd also love to have some way to download all the logs for an environment. It's real painful to use the scroll through, since it moves as new logs come in.

**Comment by timkelty** [comment]
Improving logs is definitely high on our list for improvements - we realize it is far from ideal. We will be publishing a roadmap soon, so you'll have a better idea of when to expect improvements.

Stay tuned!
